examples/qtsupport/qt_group_chat.cpp

```diff
@@ -26,57 +26,62 @@ CAF_PUSH_WARNINGS
 #include "ui_chatwindow.h" // auto generated from chatwindow.ui
 CAF_POP_WARNINGS
 
+#include "chatwidget.hpp"
+
 using namespace std;
 using namespace caf;
 
 class config : public actor_system_config {
 public:
-  std::string name;
-  std::string group_id;
-
-  config(int argc, char** argv) {
+  config() {
     opt_group{custom_options_, "global"}
-    .add(name, "name,n", "set name")
-    .add(group_id, "group,g", "join group (format: <module>:<id>");
-    parse(argc, argv);
-    load<io::middleman>();
+      .add<std::string>("name,n", "set name")
+      .add<std::string>("group,g", "join group");
   }
 };
 
 int main(int argc, char** argv) {
-  config cfg{argc, argv};
-  actor_system system{cfg};
-  auto name = cfg.name;
+  init_global_meta_objects<id_block::qt_support>();
+  core::init_global_meta_objects();
+  io::middleman::init_global_meta_objects();
+  config cfg{};
+  auto err = cfg.parse(argc, argv);
+  if (cfg.cli_helptext_printed)
+    return 0;
+  cfg.load<io::middleman>();
+  actor_system sys{cfg};
+  std::string name;
+  if (auto config_name = get_if<std::string>(&cfg, "name"))
+    name = *config_name;
+  while (name.empty()) {
+    std::cout << "please enter your name: " << std::flush;
+    if (!std::getline(std::cin, name)) {
+      std::cerr << "*** no name given... terminating" << std::endl;
+      return EXIT_FAILURE;
+    }
+  }
   group grp;
   // evaluate group parameters
-  if (! cfg.group_id.empty()) {
-    auto p = cfg.group_id.find(':');
-    if (p == std::string::npos) {
-      cerr << "*** error parsing argument " << cfg.group_id
-         << ", expected format: <module_name>:<group_id>";
+  if (auto locator = get_if<std::string>(&cfg, "group")) {
+    if (auto grp_p = sys.groups().get(*locator)) {
```

Comment:
```suggestion
    if (auto grp_p = sys.groups().get(*locator)) {
```

What is the `_p` suffix standing for? In other places, we usually use a `maybe_` prefix for catching the result of something that returns `expected<T>` or `optional<T>`.